### PR TITLE
string optimization

### PR DESCRIPTION
**Pull request by philippe44**
This makes a better use of substr for AAC & FLAC parsing, which can be pretty CPU taxing, by doing in-place string shortening + extraction instead of letting Perl do copies

### CODE REVIEW

File: Slim/Formats/Movie.pm [hunk 1]
```diff
@@ -384,7 +384,7 @@ sub extractADTS {
 		$consumed += $frame_size;
 	}	
 
-	$codec->{inbuf} = substr($codec->{inbuf}, $consumed);
+	substr($codec->{inbuf}, 0, $consumed, '');
```
Comment (michaelherger) [inline Slim/Formats/Movie.pm:387]: I didn't find documented that `substr` would do in-place replacements? This wouldn't change anything, would it?

Comment (philippe44) [reply Slim/Formats/Movie.pm:387]: I took it from there https://docstore.mik.ua/orelly/perl4/prog/ch24_02.htm and so far it seems to work on my plugins (YT) and on this PR as well (and it's way faster on a simple example that simply chew-in a large string).

When you do $a = substr($a, 10), it seems that Perl makes a copy of the whole string and then copies it back where when you do substr($a, 0, 10, ''), the first 10 bytes are removed, so I assume that something like only an offset to a pointer is changed, at least up to some garbage collector is triggered (a bit like Rust does with slices). But I have to say I have no knowledge of the usual internat Perl data representation

Comment (mherger) [reply Slim/Formats/Movie.pm:387]: Oh my... I totally misunderstood things. For whatever reason I read `substr` as if it were a replacement action mainly...

**Review by mherger — APPROVED** [review]
It's a bit of perlish magic, in particular line 1046 in FLAC.pm. But it's looking good to me anyway.

**Pull request merged**